packages/types/test/unit/phase0/sszTypes.test.ts

```diff
@@ -0,0 +1,24 @@
+import {ContainerType} from "@chainsafe/ssz";
+import {describe, it, expect} from "vitest";
+import {ssz} from "../../../src/index.js";
+import {ValidatorType} from "../../../src/phase0/validator.js";
+
+const ValidatorContainer = new ContainerType(ValidatorType, {typeName: "Validator", jsonCase: "eth2"});
+
+describe("Validator ssz types", function () {
+  it("should serialize to the same value", () => {
+    const validator = ValidatorContainer.defaultValue();
```

Comment:
Just an idea, it might be useful to directly provide overrides to `defaultValue`

```ts
const validator = ValidatorContainer.defaultValue({activationEligibilityEpoch: 10, ...});
```
and missing values use defaults

Reply:
I failed to do it as `ValidatorContainer.defaultValue()` does not support any params. We have `generateValidators()` test util but it's in beacon-node, it's not worth to do a refactor just for this I guess.

I refactored a little bit to test all possible values in validator type

Reply:
> I failed to do it as ValidatorContainer.defaultValue() does not support any params.

yeah I don't think it does, this was more of an idea we could implement in ssz library to make generating testing data a bit easier